### ISSUE TITLE
OCGV: If items are selected, then a filter is applied, items now hidden stay selected

### ISSUE THREAD

**Issue by tig** [PowerShell/GraphicalTools#121]
What does Filter do?

1) Filter means Filter DISPLAY - Restrict what you see (current implementation).
2) Filter means Filter INPUT - Restrict the data you are working with.

Example:

```powershell
Get-PSProfile | Out-ConsoleGridView -OutputMode Multiple -Title 'PS Profiles'
```

Select the two  items:
![image](https://user-images.githubusercontent.com/585482/94961804-3f6d2100-04aa-11eb-8966-b8a525dfc719.png)
![image](https://user-images.githubusercontent.com/585482/94961947-80653580-04aa-11eb-8f76-2e96aeb82e12.png)

Change the filter to "ISE":
![image](https://user-images.githubusercontent.com/585482/94961833-4e53d380-04aa-11eb-8a20-6bbd6a926b13.png)

Select the first ISE item:
![image](https://user-images.githubusercontent.com/585482/94962032-a5f23f00-04aa-11eb-94d2-2304e526c75b.png)

Hit `ENTER` to Accept.

Note three items are returned:
![image](https://user-images.githubusercontent.com/585482/94962072-b6a2b500-04aa-11eb-83ea-196089a2a94b.png)

Thus, today, the Filter only impacts what is displayed. 

Is this the right behavior? 

I could argue it either way, but being explicit is going to be more important as ocgv gains more functionality (e.g. See my Select-All issue #120). I also have dreams of a cmdlet that makes editing data easy via console GUI. It seems to me we'd want the model to be consistent (or at least explicitly inconsistent). 

Note, `ogv` works differently. If you select things in `-OutputMode Multiple` mode, and then change the filter, those selections are cleared. One could also argue that `ocgv` should match `ogv` here, which means it should use "Filter Means Filter INPUT" (2).

**Comment by corbob** [comment]
Here's my reminder to myself to read the whole thing before I start formulating my response 😬 I went and tested Windows PowerShell's `ogv` implementation to find exactly as you did that it means option 2. This is definitely the behavior I would expect and I would advocate for.

As a side note: @tig is `Get-PSProfile`  a custom function, or is it part of a module that someone might be able to download?

**Comment by tig** [comment]
`Get-PSProfile` is part of https://github.com/jdhitsolutions/PSScriptTools.

I forgot how I found it, but use it for testing because it gives lots of columns with different lengths and thus stresses ocgv well. I forgot it wasn't a built-in. I'll fix my PR to use something else built-in in the build tools.

I'll also do a PR to "fix" ocgv per this Issue...

**Comment by tig** [comment]
I believe this should be fixed. Working on PR now.

**Comment by tig** [comment]
Fixed in #166
